src/commands.cc

```diff
@@ -1485,6 +1485,35 @@ const CommandDesc debug_cmd = {
             write_to_debug_buffer(format(" * {}:\n{}",
                                   parser[1], dump_regex(compile_regex(parser[1], RegexCompileFlags::Optimize))));
         }
+        else if (parser[0] == "registers")
+        {
+            write_to_debug_buffer("Register info:");
+            const auto& register_manager = RegisterManager::instance();
+            for (Codepoint c : StringView{"\"/@^|%.#_:\\0123456789abcdefghijklmnopqrstuvwxyz"})
+            {
+                bool first = true;
+                String s;
+
+                for (String val : register_manager[c].get(context))
```

Comment:
This whole loop looks like it could be written `String s = join(reg->get(context), "\n     = ");` However I think we should also quote the contents to make the output less ambiguous (like if some element is the empty string), so it go with something like:

```
auto content = reg->get(context);
if (content.size() == 1 and content[0] == "") // empty register always return a single empty string
    continue;
write_to_debug_buffer(format("* {} = {}\n", c, join(content | transform(quote), "\n     = ")));
```

Reply:
I opted to make a "custom" quote transform to avoid the `double_up("'")` in the `quote` transform (if any register's value is `'`, it would actually display as `''`, e.g. ` * a = ''''` rather than ` * a = '''`).